File: internal/models/deployment.go
package models

import "time"

type Deployment struct {
	ID            string    `json:"id"`
	ServiceID     string    `json:"serviceId"`
	EnvironmentID string    `json:"environmentId"`
	ProjectID     string    `json:"projectId"`
	Status        string    `json:"status"`
	Branch        string    `json:"branch,omitempty"`
	CommitHash    string    `json:"commitHash,omitempty"`
	CommitMessage string    `json:"commitMessage,omitempty"`
	Trigger       string    `json:"trigger,omitempty"`
	BuildLogs     string    `json:"buildLogs,omitempty"`
	ContainerID   string    `json:"containerId,omitempty"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
	FinishedAt    time.Time `json:"finishedAt,omitempty"`
}

type ServiceMetric struct {
	Timestamp  string  `json:"timestamp"`
	CPUPercent float64 `json:"cpuPercent"`
	MemoryMB   float64 `json:"memoryMB"`
	NetworkRx  float64 `json:"networkRxKB"`
	NetworkTx  float64 `json:"networkTxKB"`
}

type TriggerDeploymentRequest struct {
	Branch *string `json:"branch,omitempty"`
}

type AppService struct {
	ID            string    `json:"id"`
	ProjectID     string    `json:"projectId"`
	EnvironmentID string    `json:"environmentId"`
	Name          string    `json:"name"`
	RepositoryURL string    `json:"repositoryUrl"`
	Branch        string    `json:"branch"`
	InternalPort  int       `json:"internalPort"`
	Domain        string    `json:"domain"`
	ContainerID   string    `json:"containerId"`
	Status        string    `json:"status"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

type CreateAppServiceRequest struct {
	ProjectID     string `json:"projectId"`
	Name          string `json:"name"`
	RepositoryURL string `json:"repositoryUrl"`
	Branch        string `json:"branch"`
	InternalPort  int    `json:"internalPort"`
	Domain        string `json:"domain"`
}

type UpdateAppServiceRequest struct {
	Name          string `json:"name"`
	RepositoryURL string `json:"repositoryUrl"`
	Branch        string `json:"branch"`
	InternalPort  int    `json:"internalPort"`
	Domain        string `json:"domain"`
	ContainerID   string `json:"containerId"`
	Status        string `json:"status"`
}

type Variable struct {
	ID            string    `json:"id"`
	ServiceID     string    `json:"serviceId"`
	ProjectID     string    `json:"projectId"`
	EnvironmentID string    `json:"environmentId"`
	Key           string    `json:"key"`
	Value         string    `json:"value"`
	IsSecret      bool      `json:"isSecret"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

type CreateServiceVarRequest struct {
	Key      string `json:"key"`
	Value    string `json:"value"`
	IsSecret bool   `json:"isSecret"`
}

type UpdateServiceVarRequest struct {
	Key      string `json:"key"`
	Value    string `json:"value"`
	IsSecret bool   `json:"isSecret"`
}

type Job struct {
	ID         string     `json:"id"`
	ProjectID  string     `json:"projectId"`
	Name       string     `json:"name"`
	Schedule   string     `json:"schedule"`
	Command    string     `json:"command"`
	Status     string     `json:"status"`
	LastRunAt  *time.Time `json:"lastRunAt"`
	LastOutput string     `json:"lastOutput"`
	CreatedAt  time.Time  `json:"createdAt"`
	UpdatedAt  time.Time  `json:"updatedAt"`
}

type CreateJobRequest struct {
	ProjectID string `json:"projectId"`
	Name      string `json:"name"`
	Schedule  string `json:"schedule"`
	Command   string `json:"command"`
}

type UpdateJobRequest struct {
	Name     *string `json:"name,omitempty"`
	Schedule *string `json:"schedule,omitempty"`
	Command  *string `json:"command,omitempty"`
	Status   *string `json:"status,omitempty"`
}

type BackupConfig struct {
	ID              string `json:"id"`
	ProjectID       string `json:"projectId"`
	DatabaseID      string `json:"databaseId,omitempty"`
	StorageID       string `json:"storageId,omitempty"`
	S3DestinationID string `json:"s3DestinationId,omitempty"`
	Name            string `json:"name"`
	Schedule        string `json:"schedule"`
	RetentionDays   int    `json:"retentionDays"`
	Status          string `json:"status"`
	CreatedAt       string `json:"createdAt"`
	UpdatedAt       string `json:"updatedAt"`
}

type BackupRecord struct {
	ID             string `json:"id"`
	BackupConfigID string `json:"backupConfigId"`
	ProjectID      string `json:"projectId"`
	DatabaseID     string `json:"databaseId,omitempty"`
	Status         string `json:"status"`
	FilePath       string `json:"filePath"`
	FileSizeBytes  int64  `json:"fileSizeBytes"`
	S3URL          string `json:"s3Url,omitempty"`
	Logs           string `json:"logs"`
	StartedAt      string `json:"startedAt"`
	CompletedAt    string `json:"completedAt"`
}

type S3Destination struct {
	ID              string `json:"id"`
	ProjectID       string `json:"projectId"`
	Name            string `json:"name"`
	Endpoint        string `json:"endpoint"`
	Bucket          string `json:"bucket"`
	Region          string `json:"region"`
	AccessKeyID     string `json:"accessKeyId"`
	SecretAccessKey string `json:"secretAccessKey"`
	CreatedAt       string `json:"createdAt"`
}

File: dashboard/src/interfaces/deployment.ts
export interface Deployment {
  id: string;
  serviceId: string;
  environmentId: string;
  projectId: string;
  status: string;
  branch?: string;
  commitHash?: string;
  commitMessage?: string;
  trigger?: string;
  buildLogs?: string;
  containerId?: string;
  createdAt: string;
  updatedAt: string;
  finishedAt?: string;
}

export interface ServiceMetric {
  timestamp: string;
  cpuPercent: number;
  memoryMB: number;
  networkRxKB: number;
  networkTxKB: number;
}

export interface TriggerDeploymentRequest {
  branch?: string;
}

export interface AppService {
  id: string;
  projectId: string;
  environmentId: string;
  name: string;
  repositoryUrl: string;
  branch: string;
  rootDirectory: string;
  buildCommand: string;
  startCommand: string;
  dockerfilePath: string;
  buildEngine: string;
  internalPort: number;
  domain: string;
  healthCheckPath: string;
  containerId: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAppServiceRequest {
  projectId: string;
  name: string;
  repositoryUrl: string;
  branch: string;
  rootDirectory: string;
  buildCommand: string;
  startCommand: string;
  dockerfilePath: string;
  buildEngine: string;
  internalPort: number;
  domain: string;
  healthCheckPath: string;
}

export interface UpdateAppServiceRequest {
  name: string;
  repositoryUrl: string;
  branch: string;
  rootDirectory: string;
  buildCommand: string;
  startCommand: string;
  dockerfilePath: string;
  buildEngine: string;
  internalPort: number;
  domain: string;
  healthCheckPath: string;
  containerId: string;
  status: string;
}

export interface Variable {
  id: string;
  serviceId: string;
  projectId: string;
  environmentId: string;
  key: string;
  value: string;
  isSecret: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateServiceVarRequest {
  key: string;
  value: string;
  isSecret: boolean;
}

export interface UpdateServiceVarRequest {
  key: string;
  value: string;
  isSecret: boolean;
}

export interface Job {
  id: string;
  projectId: string;
  name: string;
  schedule: string;
  command: string;
  status: string;
  lastRunAt: string;
  lastOutput: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateJobRequest {
  projectId: string;
  name: string;
  schedule: string;
  command: string;
}

export interface UpdateJobRequest {
  name?: string;
  schedule?: string;
  command?: string;
  status?: string;
}

export interface BackupConfig {
  id: string;
  projectId: string;
  databaseId?: string;
  storageId?: string;
  s3DestinationId?: string;
  name: string;
  schedule: string;
  retentionDays: number;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface BackupRecord {
  id: string;
  backupConfigId: string;
  projectId: string;
  databaseId?: string;
  status: string;
  filePath: string;
  fileSizeBytes: number;
  s3Url?: string;
  logs: string;
  startedAt: string;
  completedAt: string;
}

export interface S3Destination {
  id: string;
  projectId: string;
  name: string;
  endpoint: string;
  bucket: string;
  region: string;
  accessKeyId: string;
  secretAccessKey: string;
  createdAt: string;
}

export interface PRPreview {
  id: string;
  serviceId: string;
  projectId: string;
  prNumber: number;
  branch: string;
  commitHash: string;
  status: string;
  previewDomain: string;
  containerId: string;
  createdAt: string;
  updatedAt: string;
}
